src/client/game/systems/SceneryManager.ts

```
@@ -159,18 +188,44 @@ export class SceneryManager {
+        const buildingsByMaterial = new Map<number, BuildingDescriptor[]>();
         for (const b of buildings) {
-            const matIdx = Math.floor(this.random() * this.buildingMaterials.length);
+            const group = buildingsByMaterial.get(b.materialIndex) ?? [];
+            group.push(b);
+            buildingsByMaterial.set(b.materialIndex, group);
+        }
+
+        const matrix = new THREE.Matrix4();
+
+        for (const [matIdx, group] of buildingsByMaterial) {
+            if (group.length === 0) continue;
```

Biome (2.4.4)

[error] Block statements are preferred in this position.

(lint/style/useBlockStatements)